Sources/Plasma/FeatureLib/pfPython/pyImageLibModGlue.cpp

```diff
@@ -0,0 +1,159 @@
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+
+
+#include <Python.h>
+#include "pyKey.h"
+
+#include "pyImageLibMod.h"
+#include "pyImage.h"
+#include "plGImage/plMipmap.h"
+
+// glue functions
+PYTHON_CLASS_DEFINITION(ptImageLibMod, pyImageLibMod);
+
+PYTHON_DEFAULT_NEW_DEFINITION(ptImageLibMod, pyImageLibMod)
+PYTHON_DEFAULT_DEALLOC_DEFINITION(ptImageLibMod)
+
+PYTHON_INIT_DEFINITION(ptImageLibMod, args, keywords)
+{
+    PyObject* keyObj = nullptr;
+    if (!PyArg_ParseTuple(args, "O", &keyObj))
+    {
+        PyErr_SetString(PyExc_TypeError, "__init__ expects a ptKey");
+        PYTHON_RETURN_INIT_ERROR;
+    }
+    if (!pyKey::Check(keyObj))
+    {
+        PyErr_SetString(PyExc_TypeError, "__init__ expects a ptKey");
+        PYTHON_RETURN_INIT_ERROR;
+    }
+    pyKey* key = pyKey::ConvertFrom(keyObj);
+    self->fThis->setKey(*key);
+    PYTHON_RETURN_INIT_OK;
+}
+
+PYTHON_METHOD_DEFINITION(ptImageLibMod, getImage, args)
+{
+    ST::string name;
+    if (!PyArg_ParseTuple(args, "O&", PyUnicode_STStringConverter, &name)) {
+        PyErr_SetString(PyExc_TypeError, "getImage expects an image name");
+        PYTHON_RETURN_ERROR;
+    }
+
+    return self->fThis->GetImage(name);
+}
+
+PYTHON_METHOD_DEFINITION_NOARGS(ptImageLibMod, getImages)
+{
+    const std::vector<PyObject*> imageList = self->fThis->GetImages();
+    PyObject* retVal = PyTuple_New(imageList.size());
+    for (size_t curKey = 0; curKey < imageList.size(); curKey++)
+        PyTuple_SET_ITEM(retVal, curKey, imageList[curKey]);
+    return retVal;
+}
+
+PYTHON_METHOD_DEFINITION_NOARGS(ptImageLibMod, getNames)
+{
+    std::vector<ST::string> nameList = self->fThis->GetImageNames();
+    PyObject* retVal = PyTuple_New(nameList.size());
+    for (size_t curKey = 0; curKey < nameList.size(); curKey++)
+        PyTuple_SET_ITEM(retVal, curKey, PyUnicode_FromSTString(nameList[curKey]));
+    return retVal;
+}
+
+PYTHON_START_METHODS_TABLE(ptImageLibMod)
+    PYTHON_METHOD(ptImageLibMod, getImage, "Params: name\nReturns the ptImage with the specified name"),
+    PYTHON_METHOD_NOARGS(ptImageLibMod, getImages, "Returns a tuple of the library's ptImages"),
+    PYTHON_METHOD_NOARGS(ptImageLibMod, getNames, "Returns the list of image names"),
+PYTHON_END_METHODS_TABLE;
+
+// Type structure definition
+#define ptImageLibMod_AS_NUMBER       PYTHON_NO_AS_NUMBER
+#define ptImageLibMod_AS_SEQUENCE     PYTHON_NO_AS_SEQUENCE
+#define ptImageLibMod_AS_MAPPING      PYTHON_NO_AS_MAPPING
+#define ptImageLibMod_STR             PYTHON_NO_STR
+#define ptImageLibMod_GETATTRO        PYTHON_NO_GETATTRO
+#define ptImageLibMod_SETATTRO        PYTHON_NO_SETATTRO
+#define ptImageLibMod_RICH_COMPARE    PYTHON_NO_RICH_COMPARE
+#define ptImageLibMod_GETSET          PYTHON_NO_GETSET
+#define ptImageLibMod_BASE            PYTHON_NO_BASE
+PLASMA_CUSTOM_TYPE(ptImageLibMod, "Params: ilmKey\nPlasma image library modifier class");
```

Comment:
We should probably just use

```suggestion
PLASMA_DEFAULT_TYPE(ptImageLibMod, "Params: ilmKey\nPlasma image library modifier class");
```

since we don't provide any of the fancy stuff.

Reply:
Well sure, *now* we don't... ;)